src/main/java/core/basesyntax/service/impl/ReportGeneratorImpl.java

```diff
@@ -0,0 +1,20 @@
+package core.basesyntax.service.impl;
+
+import core.basesyntax.service.ReportGenerator;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+public class ReportGeneratorImpl implements ReportGenerator {
+    private static final String HEADER = "fruit" + "," + "quantity";
+
+    @Override
+    public String generateReport(Map<String, Integer> inventoryData) {
+        List<String> reportLines = new ArrayList<>();
+        reportLines.add(HEADER);
+        for (Map.Entry<String, Integer> entry: inventoryData.entrySet()) {
+            reportLines.add(entry.getKey() + "," + entry.getValue());
```

Comment:
Consider handling special characters in fruit names to ensure proper CSV formatting. This can be done by escaping or quoting values as necessary.